tm2/pkg/cmap/cmap_test.go

```diff
@@ -56,6 +58,61 @@ func TestContains(t *testing.T) {
 	assert.Nil(t, cmap.Get("key2"))
 }
 
+var sink any = nil
+
+func BenchmarkCMapConcurrentInsertsDeletesHas(b *testing.B) {
+	cm := NewCMap()
+	keys := make([]string, 100000)
+	for i := range keys {
+		keys[i] = fmt.Sprintf("key%d", i)
+	}
+	b.ResetTimer()
+
+	for i := 0; i < b.N; i++ {
+		var wg sync.WaitGroup
+		semaCh := make(chan bool)
+		nCPU := runtime.NumCPU()
+		for j := 0; j < nCPU; j++ {
+			wg.Add(1)
+			go func() {
+				defer wg.Done()
+
+				// Make sure that all the goroutines run at the
+				// exact same time for true concurrent tests.
+				<-semaCh
+
+				for i, key := range keys {
+					if (j+i)%2 == 0 {
+						cm.Has(key)
+					} else {
+						cm.Set(key, j)
+					}
+					_ = cm.Size()
+					if (i+1)%3 == 0 {
+						cm.Delete(key)
+					}
+
+					if (i+1)%327 == 0 {
+						cm.Clear()
+					}
+					_ = cm.Size()
+					_ = cm.Keys()
+				}
+				_ = cm.Values()
+			}()
+		}
+		close(semaCh)
+		wg.Wait()
+
+		sink = wg
```

Comment:
why put `sink = wg` here?

does it not maybe make sense to put a `sink` in the goroutine function, and use that as a sink for assigning the values from size, keys and values, rather than using the blank ident?

Reply:
That becomes a data race then which is definitely not something we want, I am how going to set it to semaCh